frontend/src/metabase/metadata/pages/DataModel/components/TablePicker/TableVisibilityToggle.tsx

```diff
@@ -0,0 +1,69 @@
+import { t } from "ttag";
+
+import { useUpdateTableMutation } from "metabase/api";
+import { useToast } from "metabase/common/hooks";
+import { Icon } from "metabase/ui";
+import type { Table } from "metabase-types/api";
+
+export function TableVisibilityToggle({
+  table,
+  className,
+}: {
+  table?: Table;
+  className?: string;
+}) {
+  const [updateTable] = useUpdateTableMutation();
+  const [sendToast] = useToast();
+
+  if (!table) {
+    return null;
+  }
+
+  const isHidden = table?.visibility_type === "hidden";
+
+  return (
+    <Icon
+      name={isHidden ? "eye_crossed_out" : "eye"}
+      className={className}
+      onClick={async (evt) => {
```

Comment:
nit 1: perhaps a callback would be easier to read.
nit 2: maybe use `ActionIcon` instead?